### PR TITLE
Make it possible for providers to change IParameterNameGeneratorFactory service

### PR DESCRIPTION
**Pull request by ajcvickers**
Per request from Oracle to generate Oracle-specific parameter names. To use, add an override something like this in your XxxDatabaseProviderServices class:

```
public override IParameterNameGeneratorFactory ParameterNameGeneratorFactory => GetService<XxxParameterNameGeneratorFactory>();
```

And register XxxParameterNameGeneratorFactory in your AddXxx method.

### CODE REVIEW

File: src/EntityFramework.Relational/RelationalDatabaseProviderServices.cs [hunk 1]
```diff
@@ -27,13 +27,15 @@ protected RelationalDatabaseProviderServices([NotNull] IServiceProvider services
...
         public override IValueGeneratorSelector ValueGeneratorSelector => GetService<RelationalValueGeneratorSelector>();
+        public override IModelValidator ModelValidator => GetService<RelationalModelValidator>();
+
         public virtual IRelationalTypeMapper TypeMapper => GetService<RelationalTypeMapper>();
         public virtual IMigrationAnnotationProvider MigrationAnnotationProvider => GetService<MigrationAnnotationProvider>();
         public virtual IBatchExecutor BatchExecutor => GetService<BatchExecutor>();
         public virtual IRelationalValueBufferFactoryFactory ValueBufferFactoryFactory => GetService<TypedValueBufferFactoryFactory>();
         public virtual ICommandBatchPreparer CommandBatchPreparer => GetService<CommandBatchPreparer>();
         public virtual ISqlStatementExecutor SqlStatementExecutor => GetService<SqlStatementExecutor>();
-        public override IModelValidator ModelValidator => GetService<RelationalModelValidator>();
```
Comment (divega) [inline src/EntityFramework.Relational/RelationalDatabaseProviderServices.cs]: Is this removal related?

**Comment by divega** [comment]
:shipit

File: src/EntityFramework.Relational/RelationalDatabaseProviderServices.cs (hunk 1 above)
Comment (ajcvickers) [inline src/EntityFramework.Relational/RelationalDatabaseProviderServices.cs]: It's not removed. It's just moved up a few lines to be with all the other overrides.

Comment (divega) [inline src/EntityFramework.Relational/RelationalDatabaseProviderServices.cs]: D'oh! missed it.
